### PR TITLE
POC for synchronous error rendering

### PR DESCRIPTION
**Pull request by dcramer**
High level:

- Expose `Spotlight.trigger(eventName, eventPayload)`
- Create "sentry:showError" event (expection is integrations namespace events).
- Trigger "sentry:showError" with eventId and event to synchronously render current errors.

### CODE REVIEW

File: demos/astro-playground/sentry.client.config.ts [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
+import { getCurrentHub, type Event } from '@sentry/astro';
 
 Sentry.init({
   debug: true,
   integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
   tracesSampleRate: 1.0,
 });
+
+if (process.env.NODE_ENV === 'development') {
+  import('@spotlightjs/astro').then(Spotlight => {
```
Comment (dcramer) [inline demos/astro-playground/sentry.client.config.ts:11]: so we could actually add localStorage settings into spotlight and make this a default behavior for _any_ unhandled error coming in too.

can also keep this behavior (i think we NEED it) and continue w/ a setting to toggle it

**Comment by dcramer** [comment]
To ship this theres a few things, some imo required, some could be second pass:

1. Clean up the router implementation. We probably need to be a bit smarter about where all the overlay code lives vs where the rendering layer starts. We should do _all_ setup before rendering. This is still a bit all over the place, and may need abstracted via  React Context helper.

2. It'd be nice to have type safety on Spotlight.trigger.

3. We need to figure out a path to make this little hack to inject the error smarter. On server-side this is actually pretty dang easy, but we'll still need to be framework aware (ughhhhh). I do THINK the default of "pop the modal on any unhandled error" is ok but I'm 100% sure thisll be annoying with SOA...

**Comment by dcramer** [comment]
> We need to figure out a path to make this little hack to inject the error smarter. On server-side this is actually pretty dang easy, but we'll still need to be framework aware (ughhhhh). I do THINK the default of "pop the modal on any unhandled error" is ok but I'm 100% sure thisll be annoying with SOA...

I guess the tricky bit here, and what we'd probably want to do...

```
if spotlight: <inject spotlight javascript code on the Django 500.html page>
```

File: packages/overlay/src/index.tsx [hunk 2]
```diff
@@ -121,16 +142,32 @@ export async function init({
...
-      defaultEventId={defaultEventId}
-      showTriggerButton={showTriggerButton}
-      sidecar={sidecar}
-      anchor={anchor}
-    />,
+    <MemoryRouter initialEntries={[initialTab]}>
+      <App
+        integrations={initializedIntegrations}
+        fullScreen={fullScreen}
+        defaultEventId={defaultEventId}
```
Comment (Lms24) [inline packages/overlay/src/index.tsx:165]: I think we can remove this option then, right? (already marked as a removal candidate in the JSDoc)

**Review by Lms24 — COMMENTED** [review]
I think the big problem here is that we can't reliably detect what error kind of error should open Spotlight vs. which one shouldn't.

Checking `handled` won't do the trick because JS SDKs always set `handled: false` unless users manually called `captureException`.

File: packages/overlay/src/integrations/sentry/index.ts [hunk 3]
```diff
@@ -21,6 +23,21 @@ export default function sentryIntegration(options?: SentryIntegrationOptions) {
...
       addSpotlightIntegrationToSentry(options);
+
+      const spotlightEventTarget = getSpotlightEventTarget();
+
+      const onRenderError = (e: CustomEvent) => {
+        log('Sentry Event', e.detail.event_id);
+        if (e.detail.event) sentryDataCache.pushEvent(e.detail.event);
+        // TODO: handle async
+        openSpotlight(`/errors/${e.detail.eventId}`);
+      };
```
[the remark this answers is not in the document]
Comment (dcramer) [reply packages/overlay/src/integrations/sentry/index.ts:33]: That could work though we likely need a larger context manager. I dont think we should abstract __too__ much away from React

**Pull request merged**